uber/site_sections/registration.py

```diff
@@ -804,12 +804,18 @@ def review(self, session):
     def discount(self, session, message='', **params):
         attendee = session.attendee(params)
         if 'first_name' in params:
+            if 'overridden_price' in params:
+                if isinstance(attendee.overridden_price, str):
+                    attendee.overridden_price = -1
             if not attendee.first_name or not attendee.last_name:
                 message = 'First and Last Name are required'
-            elif not attendee.overridden_price:
+            elif attendee.overridden_price < 0:
```

Comment:
This is kind of a weird way to do this - it took me two read-throughs to understand that you were checking for < 0 because you had set it to -1 earlier if it was a string.

There's no particular reason that we have to perform this check here, since the order of error messages is basically arbitrary. So you can make things simpler by removing this `elif` statement and, instead of setting the overridden_price up above, simply return an error message right then, something like 'Discounted price is required and must be a number.'